src/main/java/com/visenze/visearch/internal/BaseViSearchOperations.java

```diff
@@ -32,22 +36,65 @@ PagedSearchResult pagify(String rawResponse, String json) {
             JsonNode node = objectMapper.readTree(json);
             List<ImageResult> result = new ArrayList<ImageResult>();
             List<ObjectSearchResult> objects = null;
-            if(node.has("result"))
-                result = deserializeListResult(rawResponse, node.get("result"), ImageResult.class);
-            else if (node.has("objects"))
-                objects = deserializeListResult(rawResponse, node.get("objects"), ObjectSearchResult.class);
-            JsonNode methodNode = node.get("method");
+            List<GroupSearchResult> groupResults = null;
+
+            if(node.has(RESULT))
+                result = deserializeListResult(rawResponse, node.get(RESULT), ImageResult.class);
+            else if (node.has(OBJECTS))
+                objects = deserializeListResult(rawResponse, node.get(OBJECTS), ObjectSearchResult.class);
+            else if (node.has(GROUP_RESULTS)) {
+                JsonNode groupResultsNode = node.get(GROUP_RESULTS) ;
```

Comment:
The issue is the "group_by" field is dynamic and can be anything. So I create a group_value in GroupSearchResult class. It is not a direct mapping to JSON. For example, JSON will return {"mpid" : "xxx", "results" : []}. "xxx" is mapped to group_value field.